### PR TITLE
Implement withdraw flow

### PR DESCRIPTION
**Pull request by nicholaspai**
Issue #1040 

First checks if user has a pending withdrawal:

If user has a pending withdrawal:
- If pending withdrawal has passed:
     - Display: pending withdrawal amount, new collateralization ratio if withdrawal were to go through
     - Prompt user whether they want to confirm or cancel this withdrawal
- If still pending:
     - Display: pending withdrawal amount, new collateralization ratio if withdrawal were to go through, when pending request will pass
     - Prompt user whether they want to cancel this withdrawal

If user does not have a pending withdrawal:
- Display: current collateralization ratio, max amount they can withdraw instantly (via `withdraw()`)
- Prompt user to enter amount they want to withdraw, which will either send a `requestWithdrawal()` or `withdraw()`

### CODE REVIEW

File: core/scripts/cli/sponsor/create.js [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
   const scalingFactor = toBN(toWei("1"));
   // TODO: Understand why we need a .rawValue in one case but not the other.
   const totalPositionCollateral = toBN((await emp.totalPositionCollateral()).rawValue.toString());
   const totalTokensOutstanding = toBN((await emp.totalTokensOutstanding()).toString());
   if (totalTokensOutstanding.isZero()) {
     // When creating the globally first position, we wouldn't have a GCR. Therefore, creating that position is a
     // different flow that isn't currently part of this tool.
-    console.log("Error: can't create initial position with this tool");
+    console.log("Error: This tool does not currently support creating the chosen market's first position");
+    return;
```
Comment (nicholaspai) [inline core/scripts/cli/sponsor/create.js]: I added this because before the CLI would crash here

File: core/scripts/cli/sponsor/redeem.js [hunk 2]
```diff
@@ -30,7 +30,8 @@ const redeem = async (web3, artifacts, emp) => {
     name: "numTokens",
     message: "How many tokens to repay, at " + fromWei(collateralPerToken) + " ETH each?",
     validate: value =>
-      (value > 0 && toWei(value) <= walletTokens) || "Number of tokens must be positive and up to your current balance"
+      (value > 0 && toBN(toWei(value)).lte(toBN(walletTokens))) ||
```
Comment (nicholaspai) [inline core/scripts/cli/sponsor/redeem.js]: I ran into errors where certain numbers would not be accepted that were below my balance. I think using BN.js addresses this change. 

For ex. I created a position with 150 collateral and 100 tokens and tried to redeem 20.

File: core/scripts/local/AdvanceEMPTime.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+/**
```
Comment (mrice32) [inline core/scripts/local/AdvanceEMPTime.js]: nit: I think this should be in a separate PR.

File: financial-templates-lib/test/ExpiringMultiPartyClient.js [hunk 4]
```diff
@@ -29,11 +30,10 @@ contract("ExpiringMultiPartyClient.js", function(accounts) {
     await collateralToken.mint(accounts[0], toWei("100000"), { from: accounts[0] });
     await collateralToken.mint(accounts[1], toWei("100000"), { from: accounts[0] });
 
-    // Create identifier whitelist and register the price tracking ticker with it.
-    identifierWhitelist = await IdentifierWhitelist.new();
+    identifierWhitelist = await IdentifierWhitelist.deployed();
```
Comment (nicholaspai) [inline financial-templates-lib/test/ExpiringMultiPartyClient.js]: This is admittedly outside the scope of the PR, but there was redundant code in adding price identifiers to the whitelist.

File: liquidator/test/Liquidator.js [hunk 5]
```diff
@@ -53,12 +53,8 @@ contract("Liquidator.js", function(accounts) {
     await collateralToken.mint(liquidatorBot, toWei("100000"), { from: contractCreator });
 
     // Create identifier whitelist and register the price tracking ticker with it.
-    identifierWhitelist = await IdentifierWhitelist.new({
```
Comment (nicholaspai) [inline liquidator/test/Liquidator.js]: Same comment as in `ExpiringMultiPartyClient`

File: core/scripts/local/DeployEMP.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
+/**
```
Comment (mrice32) [inline core/scripts/local/DeployEMP.js]: nit: I think this should also be a separate PR.

Comment (nicholaspai) [reply core/scripts/local/DeployEMP.js]: deleted

File: core/scripts/local/AdvanceEMPTime.js (hunk 3 above)
Comment (nicholaspai) [reply core/scripts/local/AdvanceEMPTime.js]: deleted

File: financial-templates-lib/test/ExpiringMultiPartyClient.js [hunk 7]
```diff
@@ -29,11 +30,10 @@ contract("ExpiringMultiPartyClient.js", function(accounts) {
...
     await identifierWhitelist.addSupportedIdentifier(web3.utils.utf8ToHex("UMATEST"));
 
-    // Create a mockOracle and finder. Register the mockMoracle with the finder.
+    // Create a mockOracle and finder. Register the mockOracle with the finder.
```
Comment (chrismaree) [inline financial-templates-lib/test/ExpiringMultiPartyClient.js]: rip to `mockMoracle`

Comment (nicholaspai) [reply financial-templates-lib/test/ExpiringMultiPartyClient.js]: the one true God

Comment (ptare) [reply financial-templates-lib/test/ExpiringMultiPartyClient.js]: hahahahahahhaha
